### PR TITLE
examples: improve storybook and add with-lokijs

### PR DESCRIPTION
**Pull request by husayt**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: examples/with-lokijs/nuxt.config.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  loading: { color: "green" },
+
+  /*
+   ** Add server middleware
+   ** Nuxt.js uses `connect` module as server
+   ** So most of express middleware works with nuxt.js server middleware
+   */
+  serverMiddleware: [
+    // body-parser middleware
+    bodyParser.json(),
```
Comment (pi0) [inline examples/with-lokijs/nuxt.config.js]: Using bodyParser for all requests slows down SSR and makes problem with Proxy. So i suggest moving it to express middleware stack.

File: examples/with-lokijs/renovate.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (pi0) [inline examples/with-lokijs/renovate.json]: I think this would not be required for example

File: examples/with-lokijs/store/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          },
+          autosave: false
+          // autosaveInterval: 4000
+        })
+      })
+      console.log("async loaded,", res.data)
+      await dispatch("LOAD_DB")
+    } else {
+      console.log(
+        "HEYYYYYYYYYYYYWHOOOOOOISCALLLINGGGGGGGGGGGGMEEEEEEEEEEEEE???????"
```
Comment (pi0) [inline examples/with-lokijs/store/index.js]: :D

File: examples/with-lokijs/store/index.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  }
+}
+
+export const actions = {
+  async nuxtServerInit({ dispatch }) {
+    console.log("STARTED SERVER")
+
+    await dispatch("INIT_DB")
+  },
+  async INIT_DB({ dispatch }) {
```
Comment (pi0) [inline examples/with-lokijs/store/index.js]: The preferred way would be creating a plugin like `loki.server.js`

File: examples/with-lokijs/.codesandbox/workspace.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{
```
Comment (pi0) [inline examples/with-lokijs/.codesandbox/workspace.json]: I think either this file is not required too.

File: examples/with-lokijs/api/index.js [hunk 6]
```diff
@@ -0,0 +1,112 @@
+console.log("API INITIAL ================================================== ")
```
Comment (pi0) [inline examples/with-lokijs/api/index.js]: No function calls should happen before imports.

File: examples/with-lokijs/store/index.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    if (process.server) {
+      const res = await new Promise(resolve => {
+        const Loki = require("lokijs")
+        db = new Loki("quickstart.db", {
```
Comment (pi0) [inline examples/with-lokijs/store/index.js]: Doing this inside nuxtServerInit or plugin function makes Loki to be loaded per each SSR request. It has to be done in a global scope.

Comment (pi0) [reply examples/with-lokijs/store/index.js]: Plugin can inject `$loki` into context so other store action would benefit from it.

**Review by pi0 — CHANGES_REQUESTED** [review]

**Comment by pi0** [comment]
Thanks @husayt for all your efforts on this. If you can split this PR into separate ones for each example that would be GREAT. There are also some lint problems to be resolved.

**Comment by husayt** [comment]
Thanks. I am currently away.
Will look into it when I get back.

**Comment by manniL** [comment]
cc @husayt

File: examples/with-lokijs/nuxt.config.js (hunk 1 above)
Comment (husayt) [reply examples/with-lokijs/nuxt.config.js]: very good point. Thanks

**Comment by husayt** [comment]
@pi0 I have taken into account all your feedback and rewritten the whole thing. [Here](https://github.com/nuxt/nuxt.js/pull/7034) is a new pull request for this.

**Pull request closed without merging**